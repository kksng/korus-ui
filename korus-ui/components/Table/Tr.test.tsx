import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Tr } from './Tr';

describe('Tr component tests:', () => {
  it('should be defined', () => {
    expect(Tr).toBeDefined();
  });
  it('should render', () => {
    const wrapper = shallow(<Tr>test</Tr>);

    expect(wrapper.text()).toEqual('test');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call onClick callback', () => {
    const handleClick = jest.fn();

    const wrapper = shallow(<Tr onClick={handleClick}>test</Tr>);

    wrapper.simulate('click');

    expect(handleClick).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
