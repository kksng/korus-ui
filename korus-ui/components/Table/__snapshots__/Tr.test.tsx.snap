// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tr component tests: should call onClick callback 1`] = `
<tr
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  test
</tr>
`;

exports[`Tr component tests: should render 1`] = `
<tr>
  test
</tr>
`;
